### PR TITLE
6 create movable desstructable object

### LINKED ISSUE #6: Create movable/destructable object
- Create movable objects
- Create destructable objects

### PR DESCRIPTION
**Pull request by Zulrage**
Closes #6 
- Handle collision
- Add defensive and offensive abilities in GenericUserData

### CODE REVIEW

File: core/src/game/shadowlight/entities/Leader.java [hunk 1]
```diff
@@ -1,33 +1,37 @@
...
-			}
-			break;
-		default:
-			return false;
-		}
-		return true;
-	}
-
-}
+package game.shadowlight.entities;
```
Comment (Xuecnahc) [inline core/src/game/shadowlight/entities/Leader.java:1]: Format file

File: core/src/game/shadowlight/entities/Adventurer.java [hunk 2]
```diff
@@ -1,139 +1,112 @@
...
+      }
+    }
+    return false;
+  }
+
+  public float getWidth() {
+    return WIDTH;
+  }
+
+  public float getHeight() {
```
Comment (Xuecnahc) [inline core/src/game/shadowlight/entities/Adventurer.java]: Check if used or not

File: core/src/game/shadowlight/entities/collidable/GameContactListener.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+public class GameContactListener implements ContactListener, ContactFilter {
+
+  @Override
+  public void beginContact(Contact contact) {
+    // No need
+  }
+
+  @Override
+  public void endContact(Contact contact) {
+    // No need
```
Comment (Xuecnahc) [inline core/src/game/shadowlight/entities/collidable/GameContactListener.java]: Use
```java
// Nothing to do
```

File: core/src/game/shadowlight/entities/collidable/ICollisionReaction.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import com.badlogic.gdx.physics.box2d.Fixture;
+import com.badlogic.gdx.physics.box2d.Manifold;
+
+import game.shadowlight.entities.type.GenericUserData;
+
+/**
+ * Interface to help handling collision within the {@link GenericUserData} when
+ * the world step
+ * 
+ * @author Bertrand
```
Comment (Xuecnahc) [inline core/src/game/shadowlight/entities/collidable/ICollisionReaction.java]: Zulrage?

File: core/src/game/shadowlight/screens/PlayScreen.java [hunk 5]
```diff
@@ -68,11 +71,20 @@ else if (player.getBody().getPosition().x > bottomRight.x)
...
     batch.begin();
     for (Body body : tmpBodies)
-      if (body.getUserData() instanceof Sprite) {
-        Sprite sprite = (Sprite) body.getUserData();
-        sprite.setPosition(body.getPosition().x - sprite.getWidth() / 2, body.getPosition().y - sprite.getHeight() / 2);
-        sprite.setRotation(body.getAngle() * MathUtils.radiansToDegrees);
-        sprite.draw(batch);
+      if (body.getUserData() instanceof GenericUserData) {
+        GenericUserData data = (GenericUserData) body.getUserData();
+        if (!data.isToBeDestroyed()) {
```
Comment (Xuecnahc) [inline core/src/game/shadowlight/screens/PlayScreen.java]: Rename to `canBeDestroyed` or `isDestroyable`

File: core/src/game/shadowlight/screens/PlayScreen.java [hunk 6]
```diff
@@ -116,8 +129,8 @@ public void show() {
 
     // Create player and set listeners
     player = new Leader(world, 0, 1, 1);
-    world.setContactFilter(player);
-    world.setContactListener(player);
+    MovableObject object = new Box(world, 1, 1, 1, 1);
```
Comment (Xuecnahc) [inline core/src/game/shadowlight/screens/PlayScreen.java]: Remove this 🐑

Comment (Xuecnahc) [reply core/src/game/shadowlight/screens/PlayScreen.java]: De aroooooouuuu

File: core/src/game/shadowlight/entities/collidable/GameContactListener.java (hunk 3 above)
Comment (Zulrage) [reply core/src/game/shadowlight/entities/collidable/GameContactListener.java]: You're really needy

File: core/src/game/shadowlight/entities/collidable/ICollisionReaction.java (hunk 4 above)
Comment (Zulrage) [reply core/src/game/shadowlight/entities/collidable/ICollisionReaction.java]: Nope, an anymous developer

File: core/src/game/shadowlight/screens/PlayScreen.java (hunk 6 above)
Comment (Zulrage) [reply core/src/game/shadowlight/screens/PlayScreen.java]: 🐐

**Comment by Xuecnahc** [comment]
👍

**Pull request merged**